floor.go

```diff
@@ -102,24 +138,90 @@ func (f *Floor) watchFloorPrice() {
 					}
 					lastUpdate.With(prometheus.Labels{"type": "floor", "ticker": f.Name, "guild": gu.Name}).SetToCurrentTime()
 					time.Sleep(time.Duration(f.Frequency) * time.Second)
+
+					if f.Color {
+						// get roles for colors
+						var redRole string
+						var greeenRole string
+
+						roles, err := dg.GuildRoles(gu.ID)
+						if err != nil {
+							logger.Errorf("Getting guilds: %s", err)
+							continue
+						}
+
+						// find role ids
+						for _, r := range roles {
+							if r.Name == "tickers-red" {
+								redRole = r.ID
+							} else if r.Name == "tickers-green" {
+								greeenRole = r.ID
+							}
+						}
+
+						if len(redRole) == 0 || len(greeenRole) == 0 {
+							logger.Error("Unable to find roles for color changes")
+							continue
+						}
+
+						// assign role based on change
+						if increase {
+							err = dg.GuildMemberRoleRemove(gu.ID, f.ClientID, redRole)
+							if err != nil {
+								logger.Errorf("Unable to remove role: %s", err)
+							}
+							err = dg.GuildMemberRoleAdd(gu.ID, f.ClientID, greeenRole)
+							if err != nil {
+								logger.Errorf("Unable to set role: %s", err)
+							}
+						} else {
+							err = dg.GuildMemberRoleRemove(gu.ID, f.ClientID, greeenRole)
+							if err != nil {
+								logger.Errorf("Unable to remove role: %s", err)
+							}
+							err = dg.GuildMemberRoleAdd(gu.ID, f.ClientID, redRole)
+							if err != nil {
+								logger.Errorf("Unable to set role: %s", err)
+							}
+						}
+					}
```

Comment:
This logic here has been moved to a function and can be replaced by a variation of what is here https://github.com/rssnyder/discord-stock-ticker/blob/master/ticker.go#L596-L602